src/main/java/org/opensearch/neuralsearch/query/NeuralSparseTwoPhaseParameters.java

```diff
@@ -0,0 +1,206 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.neuralsearch.query;
+
+import com.google.common.annotations.VisibleForTesting;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+import lombok.experimental.Accessors;
+import org.apache.commons.lang.builder.HashCodeBuilder;
+import org.opensearch.Version;
+import org.opensearch.cluster.service.ClusterService;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.core.ParseField;
+import org.opensearch.core.common.ParsingException;
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.common.io.stream.StreamOutput;
+import org.opensearch.core.common.io.stream.Writeable;
+import org.opensearch.core.xcontent.XContentBuilder;
+import org.opensearch.core.xcontent.XContentParser;
+import org.opensearch.neuralsearch.settings.NeuralSearchSettings;
+import org.opensearch.neuralsearch.util.NeuralSearchClusterUtil;
+
+import java.io.IOException;
+import java.util.Locale;
+import java.util.Objects;
+
+/**
+ * Represents the parameters for neural_sparse two-phase process.
+ * This class encapsulates settings related to window size expansion, pruning ratio, and whether the two-phase search is enabled.
+ * It includes mechanisms to update settings from the cluster dynamically.
+ */
+@Getter
+@Setter
+@Accessors(chain = true, fluent = true)
+@NoArgsConstructor
+@AllArgsConstructor
+public class NeuralSparseTwoPhaseParameters implements Writeable {
+    public static volatile Float DEFAULT_WINDOW_SIZE_EXPANSION;
+    public static volatile Float DEFAULT_PRUNING_RATIO;
+    public static volatile Boolean DEFAULT_ENABLED;
+    public static volatile Integer MAX_WINDOW_SIZE;
+    @VisibleForTesting
+    static final ParseField NAME = new ParseField("two_phase_settings");
+    @VisibleForTesting
+    static final ParseField WINDOW_SIZE_EXPANSION = new ParseField("window_size_expansion");
+    @VisibleForTesting
+    static final ParseField PRUNING_RATIO = new ParseField("pruning_ratio");
+    @VisibleForTesting
+    static final ParseField ENABLED = new ParseField("enabled");
+    private static final Version MINIMAL_SUPPORTED_VERSION_TWO_PHASE_SEARCH = Version.V_2_14_0;
+
+    private Float window_size_expansion;
+    private Float pruning_ratio;
+    private Boolean enabled;
+
+    /**
+     * Initialize when start a cluster.
+     *
+     * @param clusterService The opensearch clusterService.
+     * @param settings       The env settings to initialize.
+     */
+    public static void initialize(ClusterService clusterService, Settings settings) {
```

Comment:
please make params of public method as `final`. Same for other methods in this class

Reply:
Change all public method para as final , besides euqals(Object object)